src/broker/redis.rs

```diff
@@ -299,92 +395,134 @@ impl Broker for RedisBroker {
     async fn consume<E: Fn(BrokerError) + Send + Sync + 'static>(
         &self,
         queue: &str,
-        _handler: Box<E>,
-    ) -> Result<Self::DeliveryStream, BrokerError> { 
-        let consumer = Consumer{
-            channel: self.consume_channel.clone(),
-            queue_name: String::from(queue),
+        error_handler: Box<E>,
+    ) -> Result<Self::DeliveryStream, BrokerError> {
+        let consumer = Consumer {
+            channel: Channel {
+                connection: self.manager.clone(),
+                queue_name: queue.to_string(),
+            },
+            error_handler,
+            polled_pop: None,
+            prefetch_count: Arc::clone(&self.prefetch_count),
+            pending_tasks: Arc::clone(&self.pending_tasks),
+            waker_tx: self.waker_tx.clone(),
         };
         Ok(consumer)
     }
 
     /// Acknowledge a [`Delivery`](trait.Broker.html#associatedtype.Delivery) for deletion.
-    async fn ack(&self, _delivery: &Self::Delivery) -> Result<(), BrokerError> {
-        println!("TODO: ack");
+    async fn ack(&self, delivery: &Self::Delivery) -> Result<(), BrokerError> {
+        self.pending_tasks.fetch_sub(1, Ordering::SeqCst);
+        let (channel, delivery) = delivery;
+        channel.remove_task(delivery).await?;
+        let mut waker_rx = self.waker_rx.lock().await;
+        if let Ok(waker) = waker_rx.try_recv() {
+            waker.wake();
+        }
         Ok(())
     }
 
     /// Retry a delivery.
     async fn retry(
         &self,
-        _delivery: &Self::Delivery,
+        delivery: &Self::Delivery,
         _eta: Option<DateTime<Utc>>,
     ) -> Result<(), BrokerError> {
-        println!("TODO: retry");
+        let (channel, delivery) = delivery;
+        channel.retry_task(delivery).await?;
```

Comment:
When retrying, the pending task count must be decremented; just like it is being done in ack. Which means you might have to call the waker again too.